FileIO/MPI_IO/NodePartitionedMeshReader.cpp

```diff
@@ -71,21 +72,24 @@ MeshLib::NodePartitionedMesh* NodePartitionedMeshReader::read(
 
     MeshLib::NodePartitionedMesh* mesh = nullptr;
 
+    std::string confirmed_file_name_base = file_name_base;
+    boost::erase_last(confirmed_file_name_base, ".vtk");
```

Comment:
Use `boost::erase_last_copy()` here for simplicity.
Better to name the result as `filename_prefix` or something, because nothing is confirmed here.